app/assets/javascripts/repositories/repository_datatable.js

```diff
@@ -214,6 +214,55 @@ setTimeout(function() {
 // Enables noSearchHidden plugin
 $.fn.dataTable.defaults.noSearchHidden = true;
 
+$('form#form-export').submit(function(e) {
+  var form = this;
+
+  if (currentMode === 'viewMode') {
+    // Remove all hidden fields
+    $('#form-export').find('input[name=row_ids\\[\\]]').remove();
+    $('#form-export').find('input[name=header_ids\\[\\]]').remove();
+
+    // Append samples
+    appendSamplesIdToForm(form);
+
+    // Append visible column information
+    $('.active table#repository-table thead tr').children('th')
+                                                .each(function(i) {
+      var th = $(this);
+      var val;
+      if ($(th).attr('id') === 'checkbox' || $(th).attr('id') === 'assigned')
+        val = -1
+      else if ($(th).attr('id') === 'row-name')
+        val = -2;
+      else if ($(th).attr('id') === 'added-by')
+        val = -3;
+      else if ($(th).attr('id') === 'added-on')
+        val = -4;
+      else
+        val = th.attr('id');
+
+      if (val)
+        $(form).append(
+            $('<input>')
+            .attr('type', 'hidden')
+            .attr('name', 'header_ids[]')
+            .val(val)
+        );
+    });
+  }
+});
+
+function appendSamplesIdToForm(form) {
+  $.each(rowsSelected, function(index, rowId) {
+    $(form).append(
```

Comment:
Like I said earlier - you could reuse this part in both cases (without the loop).